app/bundles/LeadBundle/Helper/CustomFieldValueHelper.php

```diff
@@ -113,4 +113,35 @@ public static function normalize($value, $type, $properties)
 
         return $value;
     }
+
+    /**
+     * @param array<string, array<string, array<string, int|string|null>>> $fields
+     *
+     * @return array<string, string>
+     */
+    public static function anonymizationFields(array $fields): array
+    {
+        $email       = $fields['core']['email']['value'] ?? null;
+        $pos         = strpos($email, '@');
```

Comment:
What if $email is null? 

Can We replace: 

```
$pos         = strpos($email, '@');
$anonimEmail = '*'.substr($email, $pos);
```


By

```
$anonimEmail = $email ? '*' . substr($email, strpos($email, '@')) : null;
```
?

Reply:
I added a check condition